### PR TITLE
Fix Reporting Of Scavenger Adaptive Threading trc point

### PR DESCRIPTION
**Pull request by RSalman**
- Trc_MM_Scavenger_calculateRecommendedWorkingThreads_threadStallBreakDown shouldn't be reported for Concurrent Scavenger

- Replace use of `_extensions->concurrentScavenger` in Scavenger with `IS_CONCURRENT_ENABLED` macro for consistency

Signed-off-by: Salman Rana <salman.rana@ibm.com>

### CODE REVIEW

**Comment by RSalman** [comment]
@amicic fyi

File: gc/base/standard/Scavenger.cpp [hunk 1]
```diff
@@ -4606,7 +4610,7 @@ MM_Scavenger::internalGarbageCollect(MM_EnvironmentBase *envBase, MM_MemorySubSp
 	}
 
 #if defined(OMR_GC_CONCURRENT_SCAVENGER)
-	if (_extensions->concurrentScavenger && isBackOutFlagRaised()) {
+	if (IS_CONCURRENT_ENABLED && isBackOutFlagRaised()) {
```
Comment (amicic) [inline gc/base/standard/Scavenger.cpp:4613]: while here, try removing #ifdef

File: gc/base/standard/Scavenger.cpp [hunk 2]
```diff
@@ -3980,7 +3984,7 @@ MM_Scavenger::processRememberedSetInBackout(MM_EnvironmentStandard *env)
 	bool const compressed = _extensions->compressObjectReferences();
 
 #if defined(OMR_GC_CONCURRENT_SCAVENGER)
-	if (_extensions->concurrentScavenger) {
+	if (IS_CONCURRENT_ENABLED) {
```
Comment (amicic) [inline gc/base/standard/Scavenger.cpp:3987]: another #ifdef that you may try to remove

File: gc/base/standard/Scavenger.cpp (hunk 1 above)
Comment (RSalman) [reply gc/base/standard/Scavenger.cpp:4609]:
```
/gc/base/standard/Scavenger.cpp:4610:80: error: ‘J9MMCONSTANT_IMPLICIT_GC_PERCOLATE_ABORTED_SCAVENGE’ was not declared in this scope
 bool result = percolateGarbageCollect(env, subSpace, NULL, ABORTED_SCAVENGE, J9MMCONSTANT_IMPLICIT_GC_PERCOLATE_ABORTED_SCAVENGE);
```

Comment (RSalman) [reply gc/base/standard/Scavenger.cpp:4609]: `J9MMCONSTANT_IMPLICIT_GC_PERCOLATE_ABORTED_SCAVENGE` is defined under CS

**Review by amicic — APPROVED** [review]

**Comment by babsingh** [comment]
jenkins build all

**Review by babsingh — APPROVED** [review]

**Pull request merged**